src/main/kotlin/com/safehill/kclient/models/SHRemoteUser.kt

```diff
@@ -36,6 +44,39 @@ data class SHRemoteUser(
         }
     }
 
+    class MapInResultDeserializer : ResponseDeserializable<Map<String, SHRemoteUser>> {
+        private val mapSerializer = MapSerializer(
+            String.serializer(),
+            SHRemoteUser.serializer()
+        )
+
+        override fun deserialize(content: String): Map<String, SHRemoteUser> {
+            return Json.decodeFromString(
+                object : DeserializationStrategy<Map<String, SHRemoteUser>> {
+                    override val descriptor: SerialDescriptor =
+                        buildClassSerialDescriptor("SHRemoteUserResults") {
+                            element<Map<String, SHRemoteUser>>("result")
+                        }
+
+                    override fun deserialize(decoder: Decoder): Map<String, SHRemoteUser> {
+                        return decoder.decodeStructure(descriptor) {
+                            var map: Map<String, SHRemoteUser>? = null
+                            when (val index = decodeElementIndex(descriptor)) {
+                                0 -> map = decodeSerializableElement(
+                                    descriptor = mapSerializer.descriptor,
+                                    index = index,
+                                    deserializer = mapSerializer
+                                )
+                            }
+                            map!!
+                        }
+                    }
+                },
+                content,
+            )
+        }
+    }
+
     class ListDeserializer : ResponseDeserializable<List<SHRemoteUser>> {
```

Comment:
I'm not sure these serializers should be added to the model, but they rather belong (like the other serializers) to com.safehill.kclient.api.serde
What do you think?

Reply:
We probably do not need ```SHRemoteUser.Deserializer``` and  ```SHRemoteUser.ListDeserializer``` for deserializing. We can use [fuel-kotlinx-serialization](https://fuel.gitbook.io/documentation/deserialization/fuel-kotlinx-serialization) for serialization using [KotlinX Serialization](https://github.com/Kotlin/kotlinx.serialization#gradle) in fuel.

For the ```SHRemoteUser.MapInResultDeserializer``` though, we would need a deserializer as we have some parsing to do. I do not have a strong recommendation for its location. As you said, ```com.safehill.kclient.api.serde``` is a viable choice in my opinion.

Reply:
Upon some thought, I dont think we would be needing ```SHRemoteUser.MapInResultDeserializer``` as well. We could have a class as shown for deserializing.
```
@Serialization
data class ContactMatchingUsers(
val result: Map<String, SHRemoteUser>
)
```